frontend/src/tests/user-permission-tests.js

```diff
@@ -14,42 +14,142 @@ import { databaseSeeder } from "./setup";
-        const resp = await apiGET("/admin/active_user");
+        let resp = await apiGET("/admin/active_user");
         expect(resp).toHaveStatus("success");
     });
 
     it("A non-admin cannot access the admin route", async () => {
-        let resp = await apiPOST("/admin/users", newUserData);
-        // Save any extra attributes we got, such as an `id`
-        Object.assign(newUserData, resp.payload);
+        await switchToTargetUser(instructorOnlyUserData);
 
-        // Set the active user to the newly-created user
-        resp = await apiPOST("/debug/active_user", { id: newUserData.id });
+        // Try to fetch admin routes
+        let resp = await apiGET("/admin/active_user");
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiGET("/admin/instructors");
+        expect(resp).toHaveStatus("error");
 
-        // Try to fetch an admin route
-        resp = await apiGET("/admin/active_user");
+        resp = await apiGET("/admin/sessions");
         expect(resp).toHaveStatus("error");
 
-        // Set the active_user back to the default.
-        resp = await apiPOST("/debug/active_user", {
-            id: databaseSeeder.seededData.active_user.id,
-        });
+        resp = await apiGET("/admin/users");
+        expect(resp).toHaveStatus("error");
+
+        resp = await apiGET("/admin/applicants");
+        expect(resp).toHaveStatus("error");
+
+        await restoreDefaultUser();
+    });
+
+    it("An instructor can access instructor routes", async () => {
+        // the default user has all roles, including the instructor role
+        let resp = await apiGET("/instructor/instructors");
+        expect(resp).toHaveStatus("success");
+
+        // switch to instructor only user
+        await switchToTargetUser(instructorOnlyUserData);
+        resp = await apiGET("/instructor/instructors");
+        expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
+    });
+
+    it("A TA can access TA routes", async () => {
+        // the default user has all roles, including the TA role
+        let resp = await apiGET("/ta/active_user");
+        expect(resp).toHaveStatus("success");
+
+        await switchToTargetUser(TAOnlyUserData);
+
+        // Try to fetch a TA route
+        resp = await apiGET("/ta/active_user");
+        expect(resp).toHaveStatus("success");
+
+        await restoreDefaultUser();
     });
 
-    it.todo("An instructor can access instructor routes");
-    it.todo("A TA can access TA routes");
-    it.todo(
-        "An instructor who is also a TA can access instructor and TA routes"
-    );
+    it("An instructor who is also a TA can access instructor and TA routes", async () => {
+        // the default user has all roles, including both the instructor role and the TA role
+        let resp = await apiGET("/ta/active_user");
+        expect(resp).toHaveStatus("success");
+        resp = await apiGET("/instructor/instructors");
```

Comment:
Again, `/instructor/instructors` is a weird route to be checking when you have `/instructor/active_user` that you could check instead.